### PR TITLE
BACK-80

### PR DESCRIPTION
**Pull request by Manizha-khorram**
User can add image link when creating Book

### CODE REVIEW

**Comment by larasodjati** [comment]
Good job @Manizha-khorram! I tested it and I can add an external link on the body and form data for book creation, 
I think we also need to add this condition/function on the update book controller as well.

@jpremo any advice on this? 

Thank you.

File: src/controllers/AllBooksController.js [hunk 1]
```diff
@@ -178,15 +178,19 @@ const createBook = async (req, res) => {
...
     }
-    const book = await Book.create(req.body)
 
-    var y = JSON.parse(JSON.stringify(book))
-    if (y.image && y.image.buffer) {
-        delete y.image
-        y.imageURL = `${baseURL}/books/image/${book.id}`
+    let book = await Book.create(req.body)
+    if (req.body.imageLink) {
+        book.imageLink = req.body.imageLink
```
Comment (jpremo) [inline src/controllers/AllBooksController.js:184]: Let's set book.imageURL to equal book.imageLink in this case

**Review by jpremo — COMMENTED** [review]
As @larasodjati suggested this logic needs to be added to all book routes that return a book. If you would prefer you can add the logic into the bookSchema itself, or simply add it onto every route.

**Comment by Manizha-khorram** [comment]
> Good job @Manizha-khorram! I tested it and I can add an external link on the body and form data for book creation, I think we also need to add this condition/function on the update book controller as well.
> 
> @jpremo any advice on this?
> 
> Thank you.

@larasodjati Thanks a lot for your review, right now you can edit imageLink like other fields related to the book. I am not sure if I need to do any thing else!

**Comment by larasodjati** [comment]
Well done Manizha, the link for image creation works for updating the book now.

**Review by larasodjati — APPROVED** [review]

**Review by jpremo — APPROVED** [review]

**Pull request merged**